src/hooks/useCorpus.ts

```diff
@@ -1,21 +1,47 @@
-import { IConfigCorpora } from '@/interfaces'
-import { useMemo } from 'react'
-
-const useCorpus = (
-  corpora?: IConfigCorpora[],
-  corpus_id?: string,
-  watchCorpusValue?: string,
-) => {
-  return useMemo(() => {
-    const getCorpusFromId = (corpusId?: string) => {
-      const corp = corpora?.find(
-        (corpus) => corpus.corpus_import_id === corpusId,
-      )
-      return corp ? corp : null
+import { useState, useEffect, useCallback } from 'react'
+
+import { IError, ICorpus } from '@/interfaces'
+import { getCorpus } from '@/api/Corpora'
+
+const useCorpus = (id?: string) => {
+  const [corpus, setCorpus] = useState<ICorpus>()
+  const [error, setError] = useState<IError | null | undefined>()
+  const [loading, setLoading] = useState<boolean>(false)
+
+  const handleGetCorpus = useCallback(() => {
+    let ignore = false
```

Comment:
I've used it as a debounce to make sure that only the latest request's response updates the state & to prevent outdated responses from overwriting newer data, thus avoiding race conditions. Have added as a comment :)